### PR TITLE
refactor geometrytypes

### PR DESCRIPTION
**Pull request by SimonDanisch**

### CODE REVIEW

**Comment by SimonDanisch** [comment]
Needs https://github.com/JuliaGeometry/GeometryTypes.jl/pull/166

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -24,7 +22,7 @@ test:
     script:
       - mkdir $JULIA_DEPOT_PATH # Pkg.jl#325
       - julia -e 'using InteractiveUtils; versioninfo()'
-      - julia --project -e 'using Pkg; Pkg.pkg"add Makie#master StatsMakie#master MakieGallery#master GLMakie#master; gc; test"'
+      - julia --project -e 'using Pkg; Pkg.pkg"test"'
```
Comment (asinghvi17) [inline .gitlab-ci.yml]: We should definitely add MakieGallery and StatsMakie master at least, no?

Comment (SimonDanisch) [reply .gitlab-ci.yml]: No we definitely should stop with that... If MakieGallery has examples that rely on a certain version of StatsMakie, it should restrict it to that version

Comment (asinghvi17) [reply .gitlab-ci.yml]: should we not also test fixes of AbstractPlotting made to StatsMakie?  That way, we at least know that the master versions work together...also, fixes to MakieGallery won't propagate here for some time.  I don't really want to tag a release to fix a typo...

Comment (SimonDanisch) [reply .gitlab-ci.yml]: We should test StatsMakie#master in StatsMakie ;) 

> I don't really want to tag a release to fix a typo...

Why not?

**Comment by SimonDanisch** [comment]
Yay, looks like it works, besides the MakieLayout change :) Gonna merge & tag!

**Comment by asinghvi17** [comment]
FEM Mesh 3D is failing here. https://gitlab.com/JuliaGPU/AbstractPlotting-jl/-/jobs/458308608#L1243

**Comment by SimonDanisch** [comment]
Yeah, but not with the newest commit, right?

**Comment by asinghvi17** [comment]
It's failing on https://github.com/JuliaPlots/AbstractPlotting.jl/pull/340/commits/1764c950178d52f5f1b2a0d5fe06f2820a577a4b, which seems to be the latest commit?

File: .gitlab-ci.yml (hunk 1 above)
Comment (asinghvi17) [reply .gitlab-ci.yml]: well, correction - I don't want to tag a release to see if my typo fix works :D

**Pull request merged**